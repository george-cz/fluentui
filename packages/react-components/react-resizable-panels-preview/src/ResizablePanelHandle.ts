export * from './components/ResizablePanelHandle/index';
