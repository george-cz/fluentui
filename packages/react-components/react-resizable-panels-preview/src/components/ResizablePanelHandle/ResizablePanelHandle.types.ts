import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type ResizablePanelHandleSlots = {
  root: Slot<'div'>;
};

/**
 * ResizablePanelHandle Props
 */
export type ResizablePanelHandleProps = ComponentProps<ResizablePanelHandleSlots> & {};

/**
 * State used in rendering ResizablePanelHandle
 */
export type ResizablePanelHandleState = ComponentState<ResizablePanelHandleSlots>;
// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from ResizablePanelHandleProps.
// & Required<Pick<ResizablePanelHandleProps, 'propName'>>
