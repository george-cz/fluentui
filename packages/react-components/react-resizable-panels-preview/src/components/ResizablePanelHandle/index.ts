export * from './ResizablePanelHandle';
export * from './ResizablePanelHandle.types';
export * from './renderResizablePanelHandle';
export * from './useResizablePanelHandle';
export * from './useResizablePanelHandleStyles.styles';
