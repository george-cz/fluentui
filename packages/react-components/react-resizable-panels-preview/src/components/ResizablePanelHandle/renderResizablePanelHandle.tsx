/** @jsxRuntime automatic */
/** @jsxImportSource @fluentui/react-jsx-runtime */

import { assertSlots } from '@fluentui/react-utilities';
import type { ResizablePanelHandleState, ResizablePanelHandleSlots } from './ResizablePanelHandle.types';

/**
 * Render the final JSX of ResizablePanelHandle
 */
export const renderResizablePanelHandle_unstable = (state: ResizablePanelHandleState) => {
  assertSlots<ResizablePanelHandleSlots>(state);

  // TODO Add additional slots in the appropriate place
  return <state.root />;
};
