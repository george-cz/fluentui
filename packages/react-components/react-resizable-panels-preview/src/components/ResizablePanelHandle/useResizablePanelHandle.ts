import * as React from 'react';
import { getIntrinsicElementProps, slot } from '@fluentui/react-utilities';
import type { ResizablePanelHandleProps, ResizablePanelHandleState } from './ResizablePanelHandle.types';

/**
 * Create the state required to render ResizablePanelHandle.
 *
 * The returned state can be modified with hooks such as useResizablePanelHandleStyles_unstable,
 * before being passed to renderResizablePanelHandle_unstable.
 *
 * @param props - props from this instance of ResizablePanelHandle
 * @param ref - reference to root HTMLDivElement of ResizablePanelHandle
 */
export const useResizablePanelHandle_unstable = (
  props: ResizablePanelHandleProps,
  ref: React.Ref<HTMLDivElement>,
): ResizablePanelHandleState => {
  return {
    // TODO add appropriate props/defaults
    components: {
      // TODO add each slot's element type or component
      root: 'div',
    },
    // TODO add appropriate slots, for example:
    // mySlot: resolveShorthand(props.mySlot),
    root: slot.always(
      getIntrinsicElementProps('div', {
        ref,
        ...props,
      }),
      { elementType: 'div' },
    ),
  };
};
