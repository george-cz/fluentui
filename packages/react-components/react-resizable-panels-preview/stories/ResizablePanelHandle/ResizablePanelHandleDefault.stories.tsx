import * as React from 'react';
import { ResizablePanelHandle, ResizablePanelHandleProps } from '@fluentui/react-resizable-panels-preview';

export const Default = (props: Partial<ResizablePanelHandleProps>) => <ResizablePanelHandle {...props} />;
