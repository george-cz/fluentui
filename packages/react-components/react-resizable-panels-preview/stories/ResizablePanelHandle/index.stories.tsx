import { ResizablePanelHandle } from '@fluentui/react-resizable-panels-preview';

import descriptionMd from './ResizablePanelHandleDescription.md';
import bestPracticesMd from './ResizablePanelHandleBestPractices.md';

export { Default } from './ResizablePanelHandleDefault.stories';

export default {
  title: 'Preview Components/ResizablePanelHandle',
  component: ResizablePanelHandle,
  parameters: {
    docs: {
      description: {
        component: [descriptionMd, bestPracticesMd].join('\n'),
      },
    },
  },
};
