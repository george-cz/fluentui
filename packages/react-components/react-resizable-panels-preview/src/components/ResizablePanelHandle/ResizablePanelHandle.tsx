import * as React from 'react';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
import { useCustomStyleHook_unstable } from '@fluentui/react-shared-contexts';
import { useResizablePanelHandle_unstable } from './useResizablePanelHandle';
import { renderResizablePanelHandle_unstable } from './renderResizablePanelHandle';
import { useResizablePanelHandleStyles_unstable } from './useResizablePanelHandleStyles.styles';
import type { ResizablePanelHandleProps } from './ResizablePanelHandle.types';

/**
 * ResizablePanelHandle component - TODO: add more docs
 */
export const ResizablePanelHandle: ForwardRefComponent<ResizablePanelHandleProps> = React.forwardRef((props, ref) => {
  const state = useResizablePanelHandle_unstable(props, ref);

  useResizablePanelHandleStyles_unstable(state);
  // TODO update types in packages/react-components/react-shared-contexts/src/CustomStyleHooksContext/CustomStyleHooksContext.ts
  // https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/convergence/custom-styling.md
  useCustomStyleHook_unstable('useResizablePanelHandleStyles_unstable')(state);
  return renderResizablePanelHandle_unstable(state);
});

ResizablePanelHandle.displayName = 'ResizablePanelHandle';
