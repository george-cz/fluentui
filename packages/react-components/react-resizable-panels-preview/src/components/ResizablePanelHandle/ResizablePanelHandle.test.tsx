import * as React from 'react';
import { render } from '@testing-library/react';
import { isConformant } from '../../testing/isConformant';
import { ResizablePanelHandle } from './ResizablePanelHandle';

describe('ResizablePanelHandle', () => {
  isConformant({
    Component: ResizablePanelHandle,
    displayName: 'ResizablePanelHandle',
  });

  // TODO add more tests here, and create visual regression tests in /apps/vr-tests

  it('renders a default state', () => {
    const result = render(<ResizablePanelHandle>Default ResizablePanelHandle</ResizablePanelHandle>);
    expect(result.container).toMatchSnapshot();
  });
});
