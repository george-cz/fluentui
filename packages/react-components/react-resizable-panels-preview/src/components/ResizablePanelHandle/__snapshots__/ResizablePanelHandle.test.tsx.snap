// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizablePanelHandle renders a default state 1`] = `
<div>
  <div
    class="fui-ResizablePanelHandle"
  >
    Default ResizablePanelHandle
  </div>
</div>
`;
